TAScheduler/views.py

```diff
@@ -178,6 +178,77 @@ def post(request):
         # TODO: implement post
         pass
 
+class DeleteLab(View):
+    def get(request):
+        # If the user does not have a valid name, I.E. if they try to manually enter /home in the search bar,
+        # they will fail the userAllowed test and be redirected back to the login page
+        # If the user is allowed then home is rendered like normal
+        if userAllowed(request, ["SUPERVISOR"]):
+            return render(request, "deletelab.html", {"lab_list":Lab.objects.all()})
+        else:
+            return redirect("/../home/")
+
+    def post(request):
+        delete = True
+        try:
+            delete_or_submit = int(request.POST["delete"])
+        except MultiValueDictKeyError:
+            delete_or_submit = int(request.POST["submit"])
+            delete = False
+        if delete:
+            deleted_lab = Lab.objects.get(labID = delete_or_submit)
+            Lab.objects.delete(labID=delete_or_submit)
+            return render(request, "deletelab.html", {"lab_list":Lab.objects.all(), "deleted_lab": deleted_lab})
+        else:
+            return render(request, "deletelab.html", {"lab_list": Lab.objects.all()})
+
+class DeleteCourse(View):
+    def get(request):
+        # If the user does not have a valid name, I.E. if they try to manually enter /home in the search bar,
+        # they will fail the userAllowed test and be redirected back to the login page
+        # If the user is allowed then home is rendered like normal
+        if userAllowed(request, ["SUPERVISOR"]):
+            return render(request, "deletecourse.html", {"course_list":Course.objects.all()})
+        else:
+            return redirect("/../home/")
+
+    def post(request):
+        delete = True
+        try:
+            delete_or_submit = int(request.POST["delete"])
+        except MultiValueDictKeyError:
+            delete_or_submit = int(request.POST["submit"])
+            delete = False
+        if delete:
+            deleted_course = Course.objects.get(courseID = delete_or_submit)
+            Course.objects.delete(courseID=delete_or_submit)
+            return render(request, "deletecourse.html", {"course_list":Course.objects.all(), "deleted_course": deleted_course})
+        else:
+            return render(request, "deletecourse.html", {"course_list": Course.objects.all()})
```

Comment:
```suggestion
        CourseManagement.deleteCourse(course_id=request.POST("delete")
        return render(request, "delete.html", {"course_list": Course.objects.all())
```
See above for more details